src/delegation/store/memory.rs

```diff
@@ -173,7 +173,7 @@ where
         // FIXME
     }
 
-    fn insert(
+    fn insert_keyed(
```

Comment:
This lets us automagically calculate the CID on `insert`, but let the trait implementer work with the lower level interface rather than forcing them to figure out the CID stuff themselves.